diesel/src/lib.rs

```diff
@@ -292,6 +291,9 @@ pub mod helper_types {
 
 pub mod prelude {
     //! Re-exports important traits and types. Meant to be glob imported when using Diesel.
+    #[doc(inline)]
+    pub use diesel_derives::*;
```

Comment:
I think the better strategy here is to put these reexports in those places where the actual traits are defined that are implemented by the corresponding custom derive. If that trait is reexported via prelude, the proc macro will also be reexported.